BigInt.h

```diff
@@ -39,23 +45,24 @@ void BigInt_assign_int(BigInt* target, const int source);
 int BigInt_compare(const BigInt* a, const BigInt* b);
 
 // Adds the value in addend to big_int.  Places the result in big_int.
-void BigInt_add(BigInt* big_int, const BigInt* addend);
-void BigInt_add_int(BigInt* big_int, const int addend);
+BOOL BigInt_add(BigInt* big_int, const BigInt* addend);
+BOOL BigInt_add_int(BigInt* big_int, const int addend);
 
 // Subtracts the value of to_subtract from big_int. 
 // Places the result in big_int.
-void BigInt_subtract(BigInt* big_int, const BigInt* to_subtract);
-void BigInt_subtract_int(BigInt* big_int, const int to_subtract);
+BOOL BigInt_subtract(BigInt* big_int, const BigInt* to_subtract);
+BOOL BigInt_subtract_int(BigInt* big_int, const int to_subtract);
 
 // Multiplies the value in big_int by multiplier.  Places the
 // result in big_int.
-void BigInt_multiply(BigInt* big_int, const BigInt* multiplier);
-void BigInt_multiply_int(BigInt* big_int, const int multiplier);
+BOOL BigInt_multiply(BigInt* big_int, const BigInt* multiplier);
+BOOL BigInt_multiply_int(BigInt* big_int, const int multiplier);
 
-// Returns the value of big_int as an integer.  Requires that the
-// value of big_int fits within the size of an int on the target
-// environment.  Result is undefined if this is not the case.
-int BigInt_to_int(const BigInt* big_int);
+// Sets result to the value of big_int as an integer if the
+// value of big_int fits within the size of result's type on the target
+// environment.  returns non-zero on success or 0 on failure.
+// result is undefined on failure.
+BOOL BigInt_to_int(int* result, const BigInt* big_int);
```

Comment:
```suggestion
BOOL BigInt_to_int(const BigInt* big_int, int* result);
```
This is nitpicky but I'd prefer to have the parameter order match the order of things like `BigInt_multiply_int` and match the fact that `BigInt` comes before `int` in the name of this function.